src/main/java/io/quarkus/search/app/fetching/FetchingService.java

```diff
@@ -103,7 +129,28 @@ private static boolean isZip(URI uri) {
                 && uri.getPath().endsWith(".zip");
     }
 
-    private <T> T gitClone(String name, URI gitUri, List<String> branches,
+    private GitCloneDirectory gitPull(GitCloneDirectory repository, Branches branches) {
+        Log.infof("Pulling changes for '%s'.", repository);
+        //String mainBranch = branches.get(0);
+        Log.infof("Pulling changes for current branch of '%s':'%s'.", repository, branches.sources());
+        try {
+            repository.git().pull()
+                    .setProgressMonitor(LoggerProgressMonitor.create(log, "Pulling '" + branches.sources() + "': "))
+                    .call();
+            repository.git().fetch().setForceUpdate(true)
+                    .setProgressMonitor(LoggerProgressMonitor.create(log, "Fetching '" + branches + "': "))
```

Comment:
I'd pass the 'name' argument here:

```suggestion
                    .setProgressMonitor(LoggerProgressMonitor.create(log, "Pulling '" + name + "' (" + branches.sources() + "): "))
                    .call();
            repository.git().fetch().setForceUpdate(true)
                    .setProgressMonitor(LoggerProgressMonitor.create(log, "Fetching '" + name + "' (" + branches + "): "))
```

Otherwise logs are quite unclear:

```
08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)
08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)
08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)
08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)
08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)

08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)

08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)

08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'main': Pull:                     50% (1/2)

08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'develop': Pull:                     50% (1/2)
08:55:15 INFO  [i.qu.se.ap.fe.FetchingService] Pulling 'develop': Pull:                     50% (1/2)

```